client/src/lib/components/Search/Search.svelte

```diff
@@ -0,0 +1,53 @@
+<script lang="ts">
+  import type { SelectOption } from "src/definitions/form";
+
+  import { createEventDispatcher } from "svelte";
+
+  const dispatch = createEventDispatcher<{ search: SelectOption }>();
+
+  export let name: string;
+  export let id = `${name}-label`;
+  export let options: Array<SelectOption> = [];
+  export let placeholder = "";
+
+  let searchInput: HTMLInputElement;
+
+  const listName = `${name}-list`;
+
+  const handleInput = (e: any) => {
+    const selectedOption = options.find(
+      (item) => item.label === e.target.value
+    );
+
+    if (selectedOption) {
+      dispatch("search", selectedOption);
+      searchInput.value = "";
+    }
+  };
+</script>
+
+<div class="fr-select-group">
+  <input
+    role="search"
+    class="fr-select"
+    {placeholder}
+    list={listName}
+    {id}
+    {name}
+    bind:this={searchInput}
+    on:input|stopPropagation|preventDefault={handleInput}
+  />
+
+  <datalist id={listName}>
```

Comment:
Dans la datalist, il semble qu'on trouve aussi les valeurs récentes :

![tags-datalist-prefill](https://user-images.githubusercontent.com/15911462/166260104-c056dd31-a06e-47d4-adb8-5ba8f3b771fd.png) (25 kB)

Pour reproduire : rentrer une valeur quelconque à la main, soumettre, puis revenir sur l'édition et constater que la valeur est dans la partie supérieure de la datalist.

On dirait qu'on peut désactiver ce comportement en ajoutant `autocomplete="off"` à l'input. https://stackoverflow.com/questions/15927182/how-to-turn-off-autocomplete-while-keep-using-datalist-element-in-html